src/twisted/logger/_format.py

```diff
@@ -287,3 +287,33 @@ def formatWithCall(formatString, mapping):
     return unicode(
         aFormatter.vformat(formatString, (), CallMapping(mapping))
     )
+
+
+
+def formatEventWithTraceback(event):
+    """
+    Format an event as a unicode string and append a traceback if the event
+    contains a C{event["log_failure"]} key.
+
+    This implementation will always attempt to append a traceback, even if
+    formatting the C{event["log_format"]} fails.  The traceback, if bytes, will
+    be assumed to be UTF-8.
+
+    @param event: A logging event.
+    @type event: L{dict}
+
+    @return: A formatted string with a traceback if applicable.
+    @rtype: L{unicode}
+    """
+    eventText = formatEvent(event)
+    try:
+        traceback = event["log_failure"].getTraceback()
+        if isinstance(traceback, bytes):
```

Comment:
I have concerns with this code.  Failure's getTraceback() method will return a native string depending on the Python version, so str/bytes Python 2.  The assumption that the bytes can sanely decoded as utf-8 is not one I've been able to verify and is probably wrong.

Looking at `formatEventAsClassicLogText`, which I used as a guide, it doesn't do any conversion but seems (I may be misreading) to rely on the fact that bytes and unicode can be joined in Python2 (https://github.com/twisted/twisted/blob/2e7b403c03e9b9b891a5eb36297e494f541d5d6a/src/twisted/logger/_format.py#L192).  That may (haven't confirmed) expose it to the same issue as 9442 due to the ascii conversion.  Should I confirm that I'll open a ticket.

So while this might be fine, I'd like to do a bit more diligence before this is merged, thus, I'm removing it from review.